client/src/index.jsx

```diff
@@ -75,37 +123,48 @@ function App() {
       });
   };
 
-  const reqErr427 = () => {
-    alert('Too many requests: wait 30-60 seconds!');
+  const switchProductPage = (e) => {
+    e.preventDefault();
+    const inputEl = document.querySelector('.search-itemId-input');
+    const inputId = parseInt(inputEl.value.trim(), 10);
+    if (!!inputId && inputId !== itemId) {
```

Comment:
never seen a double exclamation usage before,

Reply:
It converts any values to truthy or falsy. Ex: null -> !null -> false